cwms-data-api/src/main/java/cwms/cda/data/dao/location/kind/LockDao.java

```diff
@@ -238,84 +193,76 @@ static Lock map(LOCK_OBJ_T lock, double upperWarningLevel, double lowerWarningLe
-                        Double.class))
-                .withHighWaterUpperPoolWarningLevel(result.get(view.ELEV_CLOSURE_HIGH_WATER_UPPER_POOL_WARNING,
-                        Double.class))
-                .withChamberType(chamberType)
-                .withHighWaterLowerPoolWarningLevel(highWaterLowerPoolWarning)
-                .withHighWaterUpperPoolWarningLevel(highWaterUpperPoolWarning)
-                .withElevationUnits(result.get(view.ELEV_UNIT_ID))
-                .withVolumeUnits(result.get(view.VOLUME_UNIT_ID))
-                .withMaximumLockLift(result.get(view.MAXIMUM_LOCK_LIFT, Double.class))
-                .withLowWaterUpperPoolLocationLevel(lowWaterUpperPool != null
-                        ? new LockLocationLevelRef(mapToLockRef(lockLocation.getOfficeId(),
-                            String.format("%s.Elev-Closure.Inst.0.Low Water Upper Pool",
-                                    lockLocation.getName())), lowWaterUpperPool)
-                        : null)
-                .withLowWaterLowerPoolLocationLevel(lowWaterLowerPool != null
-                        ? new LockLocationLevelRef(mapToLockRef(lockLocation.getOfficeId(),
-                            String.format("%s.Elev-Closure.Inst.0.Low Water Lower Pool",
-                                    lockLocation.getName())), lowWaterLowerPool)
-                        : null)
-                .withHighWaterLowerPoolLocationLevel(highWaterLowerPool != null
-                        ? new LockLocationLevelRef(mapToLockRef(lockLocation.getOfficeId(),
-                            String.format("%s.Elev-Closure.Inst.0.High Water Lower Pool",
-                                    lockLocation.getName())), highWaterLowerPool)
-                        : null)
-                .withHighWaterUpperPoolLocationLevel(highWaterUpperPool != null
-                        ? new LockLocationLevelRef(mapToLockRef(lockLocation.getOfficeId(),
-                            String.format("%s.Elev-Closure.Inst.0.High Water Upper Pool",
-                                    lockLocation.getName())), highWaterUpperPool)
-                        : null)
-                .build();
-    }
-
     static String mapToLockRef(String office, String locationName) {
         return String.format("/locks/%s?office=%s", locationName, office);
     }
+
+    public Lock unitConvertToEN(Lock lock) {
+        return connectionResult(dsl, c -> {
+            setOffice(c, lock.getLocation().getOfficeId());
+            return new Lock.Builder()
+                .withElevationUnits("ft")
+                .withVolumeUnits("ft3")
+                .withLengthUnits("ft")
+                .withLocation(lock.getLocation())
+                .withProjectId(lock.getProjectId())
+                .withLockWidth(CWMS_UTIL_PACKAGE.call_CONVERT_UNITS(dsl.configuration(), lock.getLockWidth(),
+                        lock.getLengthUnits(), "ft"))
```

Comment:
create constants for "ft" and "ft3", or I think you could try using something like:

`CWMS_UTIL_PACKAGE.call_GET_DEFAULT_UNITS(dsl.configuration(), "Length", "EN");`
`CWMS_UTIL_PACKAGE.call_GET_DEFAULT_UNITS(dsl.configuration(), "Volume", "EN");`

Reply:
Added default unit calls